File: ios/ObjectiveC/Beacon/1-QuickStart/Complete/QuickComplete/ViewController.m
//
//  ViewController.m
//  QuickStart
//
//  Created by sarra srairi on 22/03/2016.
//  Copyright © 2016 sarra srairi. All rights reserved.
//

#import "ViewController.h"


@interface ViewController ()

@end

@implementation ViewController {
    NSString *feedStatusString;
    AdtagPlaceDetectionManager *placeDetectionManager;
    AdtagInitializer *adtagInitializer;
}

@synthesize hashValue;

- (void)viewDidLoad {
    [super viewDidLoad];
    placeDetectionManager = [AdtagPlaceDetectionManager shared];
    adtagInitializer = [AdtagInitializer shared];

    // Do any additional setup after loading the view, typically from a nib.
}

- (void) viewWillAppear:(BOOL)animated{
    // Register Range Delegate protocol to your view
    [placeDetectionManager registerInProximityInForeground:self];
    [adtagInitializer registerProximityHealthCheckDelegate:self];
    [super viewWillAppear:animated];
}

- (void) viewWillDisappear:(BOOL)animated{
    [super viewWillDisappear:animated];
    [placeDetectionManager unregisterInProximityInForeground:self];
    [adtagInitializer unregisterProximityHealthCheckDelegate:self];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void) onProximityHealthCheckUpdate:(HealthStatus *)healthStatus {
    NSMutableString *error = [[NSMutableString alloc] initWithString:@""];
    if ([healthStatus isDown]) {
        for (ServiceStatus *serviceStatus in [healthStatus.serviceStatusMap allValues]) {
            if ([serviceStatus isDown]) {
                for (Status *status in serviceStatus.statusList) {
                    [error appendString:status.message];
                    [error appendString:@"\n"];
                }
            }
        }
    }
    _txt_nbrBeacon.text = error;
}

-(void) proximityContentsInForeground:(NSArray<AdtagPlaceInAppAction *> *)contents {
    _txt_nbrBeacon.text =[NSString stringWithFormat:NSLocalizedString(@"beaconAround", @""), contents.count];
}



@end

File: ios/SWIFT/Beacon/1-QuickStart/Complete/QuickStart/ViewController.swift
//
//  ViewController.swift
//  QuickStart
//
//  Created by sarra srairi on 10/08/2016.
//  Copyright © 2016 R&D connecthings. All rights reserved.
//
import UIKit
import AdtagConnection
import ConnectPlaceActions
import AdtagLocationDetection
import ConnectPlaceCommon

class ViewController: UIViewController, AdtagInProximityInForegroundDelegate, ProximityHealthCheckDelegate {
    @IBOutlet weak var txt_message: UILabel!
    var adtagInitializer: AdtagInitializer?
    var adtagPlaceManager: AdtagPlaceDetectionManager?

    override func viewDidLoad() {
        super.viewDidLoad()
        adtagInitializer = AdtagInitializer.shared
        adtagPlaceManager = AdtagPlaceDetectionManager.shared
    }

    override func viewWillAppear(_ animated: Bool) {
        adtagInitializer?.registerProximityHealthCheckDelegate(self)
        adtagPlaceManager?.registerInProximityInForeground(self)
        super.viewWillAppear(animated)
    }

    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        adtagInitializer?.unregisterProximityHealthCheckDelegate(self)
        adtagPlaceManager?.unregisterInProximityInForeground(self)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func proximityContentsInForeground(_ contents: [AdtagPlaceInAppAction]) {
        self.txt_message.text = String( format: NSLocalizedString("beaconAround", comment:""), contents.count)
    }

    func onProximityHealthCheckUpdate(_ healthStatus: HealthStatus) {
        var error: String = ""
        if healthStatus.isDown {
            for serviceStatus in healthStatus.serviceStatusMap.values {
                if serviceStatus.isDown() {
                    for status in serviceStatus.statusList {
                        error += status.message as String + "\n"
                    }
                }
            }
        }

        self.txt_message.text = error
    }
 
}
